### ISSUE TITLE
Incorrect render function? 

### ISSUE THREAD

**Issue by ivanprotsko** [pugjs/pug#2643]
Please, tell me, why I have this problem?

Sample from the manual ( https://pugjs.org/api/reference.html#pugcompilefileclientpath-options ): 

**template.pug**
h1 This is a Pug template
h2 By #{author}

**pug-render.js**
var fs = require('fs');
var pug = require('pug');

// Compile the template to a function string
var jsFunctionString = pug.compileFileClient('/path/to/pugFile.pug', {name: "fancyTemplateFun"});

// Maybe you want to compile all of your templates to a templates.js file and serve it to the client
fs.writeFileSync("templates.js", jsFunctionString);

**Result template.js in sample**
function fancyTemplateFun(locals) {
  var buf = [];
  var pug_mixins = {};
  var pug_interp;

  var locals_for_with = (locals || {});

  (function (author) {
    buf.push("<h1>This is a Pug template</h1><h2>By "
      + (pug.escape((pug_interp = author) == null ? '' : pug_interp))
      + "</h2>");
  }.call(this, "author" in locals_for_with ?
    locals_for_with.author : typeof author !== "undefined" ?
      author : undefined)
  );

  return buf.join("");
}

**Why i got this result in my template.js?**

function pug_escape(e) {
  var a = "" + e, t = pug_match_html.exec(a);
  if (!t)return e;
  var r, c, n, s = "";
  for (r = t.index, c = 0; r < a.length; r++) {
    switch (a.charCodeAt(r)) {
      case 34:
        n = "&quot;";
        break;
      case 38:
        n = "&amp;";
        break;
      case 60:
        n = "&lt;";
        break;
      case 62:
        n = "&gt;";
        break;
      default:
        continue
    }
    c !== r && (s += a.substring(c, r)), c = r + 1, s += n
  }
  return c !== r ? s + a.substring(c, r) : s
}
var pug_match_html = /["&<>]/;
function pug_rethrow(n, e, r, t) {
  if (!(n instanceof Error))throw n;
  if (!("undefined" == typeof window && e || t))throw n.message += " on line " + r, n;
  try {
    t = t || require("fs").readFileSync(e, "utf8")
  } catch (e) {
    pug_rethrow(n, null, r)
  }
  var i = 3, a = t.split("\n"), o = Math.max(r - i, 0), h = Math.min(a.length, r + i), i = a.slice(o, h).map(function (n, e) {
    var t = e + o + 1;
    return (t == r ? "  > " : "    ") + t + "| " + n
  }).join("\n");
  throw n.path = e, n.message = (e || "Pug") + ":" + r + "\n" + i + "\n\n" + n.message, n
}
function menu(locals) {
  var pug_html = "", pug_mixins = {}, pug_interp;
  var pug_debug_filename, pug_debug_line;
  try {
    ;
    var locals_for_with = (locals || {});
    (function (author) {
      ;
      pug_debug_line = 1;
      pug_debug_filename = "build\u002Fblocks\u002Fmenu\u002Fmenu.pug";
      pug_html = pug_html + "\u003Ch1\u003E";
      ;
      pug_debug_line = 1;
      pug_debug_filename = "build\u002Fblocks\u002Fmenu\u002Fmenu.pug";
      pug_html = pug_html + "This is a Pug template\u003C\u002Fh1\u003E";
      ;
      pug_debug_line = 2;
      pug_debug_filename = "build\u002Fblocks\u002Fmenu\u002Fmenu.pug";
      pug_html = pug_html + "\u003Ch2\u003E";
      ;
      pug_debug_line = 2;
      pug_debug_filename = "build\u002Fblocks\u002Fmenu\u002Fmenu.pug";
      pug_html = pug_html + "By ";
      ;
      pug_debug_line = 2;
      pug_debug_filename = "build\u002Fblocks\u002Fmenu\u002Fmenu.pug";
      pug_html = pug_html + (pug_escape(null == (pug_interp = author) ? "" : pug_interp)) + "\u003C\u002Fh2\u003E";
    }.call(this, "author" in locals_for_with ? locals_for_with.author : typeof author !== "undefined" ? author : undefined));
  } catch (err) {
    pug_rethrow(err, pug_debug_filename, pug_debug_line);
  }
  ;
  return pug_html;
}

**Comment by jbsulli** [comment]
Yeah, the example should probably be updated. Pug compileClient functions are going to change as they address security concerns and do bug fixes. In this case, I think you want the `compileDebug: false` and `inlineRuntimeFunctions: false` options to get a closer match.

`compileDebug: false` will not print out pug line numbers and filename if there is an error but it will shrink the function size considerably.

`inlineRuntimeFunctions: false` will make it so that pug_escape and other pug functions are not in the output file but you'll want to include [the runtime](https://github.com/pugjs/pug-runtime) if you do this.
